CMakeLists.txt

```diff
@@ -183,6 +204,11 @@ set(OPENTOMB_SRCS
     src/world.h
 )
 
+# Disable warnings when using unsafe functions
+if(WIN32)
```

Comment:
will it conflict with MinGW (it will build, but will it dead define for MinGW)?

Reply:
It works fine here, I can run OpenTomb only with MINGW since, as I wrote in issue #504, there are still some issues when compiling with MSVC. This is my compilation log, made with MINGW-W64 under CYGWIN:

```
$ cmake ../opentomb -DCMAKE_TOOLCHAIN_FILE=../opentomb/cmake/Toolchain-cross-mingw32-cygwin.cmake -DFORCE_SYSTEM_FREETYPE=on
-- The C compiler identification is GNU 6.4.0
-- The CXX compiler identification is GNU 6.4.0
-- Check for working C compiler: /usr/bin/i686-w64-mingw32-gcc.exe
-- Check for working C compiler: /usr/bin/i686-w64-mingw32-gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/i686-w64-mingw32-g++.exe
-- Check for working CXX compiler: /usr/bin/i686-w64-mingw32-g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Freetype: /usr/i686-w64-mingw32/sys-root/mingw/lib/libfreetype.dll.a (found version "2.6.5")
-- MinGW Cross-Compilation
-- Found ZLIB: /usr/i686-w64-mingw32/sys-root/mingw/lib/libz.dll.a (found version "1.2.11")
-- Found PNG: /usr/i686-w64-mingw32/sys-root/mingw/lib/libpng.dll.a (found version "1.6.34")
-- Looking for include file alext.h
-- Looking for include file alext.h - found
-- Looking for include file efx.h
-- Looking for include file efx.h - found
-- Looking for include file efx-presets.h
-- Looking for include file efx-presets.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/Carlo/ot

Carlo@CarloPC ~/ot
$ make
Scanning dependencies of target lua5.3
[  0%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lapi.c.obj
[  1%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lauxlib.c.obj
[  1%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lbaselib.c.obj
[  1%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lbitlib.c.obj
[  2%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lcode.c.obj
[  2%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lcorolib.c.obj
[  2%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lctype.c.obj
[  3%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ldblib.c.obj
[  3%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ldebug.c.obj
[  4%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ldo.c.obj
[  4%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ldump.c.obj
[  4%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lfunc.c.obj
[  5%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lgc.c.obj
[  5%] Building C object extern/lua/CMakeFiles/lua5.3.dir/linit.c.obj
[  5%] Building C object extern/lua/CMakeFiles/lua5.3.dir/liolib.c.obj
[  6%] Building C object extern/lua/CMakeFiles/lua5.3.dir/llex.c.obj
[  6%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lmathlib.c.obj
[  7%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lmem.c.obj
[  7%] Building C object extern/lua/CMakeFiles/lua5.3.dir/loadlib.c.obj
[  7%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lobject.c.obj
[  8%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lopcodes.c.obj
[  8%] Building C object extern/lua/CMakeFiles/lua5.3.dir/loslib.c.obj
[  8%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lparser.c.obj
[  9%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lstate.c.obj
[  9%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lstring.c.obj
[ 10%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lstrlib.c.obj
[ 10%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ltable.c.obj
[ 10%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ltablib.c.obj
[ 11%] Building C object extern/lua/CMakeFiles/lua5.3.dir/ltm.c.obj
[ 11%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lundump.c.obj
[ 11%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lutf8lib.c.obj
[ 12%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lvm.c.obj
[ 12%] Building C object extern/lua/CMakeFiles/lua5.3.dir/lzio.c.obj
[ 13%] Linking C static library liblua5.3.a
[ 13%] Built target lua5.3
Scanning dependencies of target bullet
[ 14%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btAxisSweep3.cpp.obj
[ 14%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btBroadphaseProxy.cpp.obj
[ 14%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btCollisionAlgorithm.cpp.obj
[ 15%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btDbvt.cpp.obj
[ 15%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btDbvtBroadphase.cpp.obj
[ 16%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btDispatcher.cpp.obj
[ 16%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btMultiSapBroadphase.cpp.obj
[ 16%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btOverlappingPairCache.cpp.obj
[ 17%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btQuantizedBvh.cpp.obj
[ 17%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/BroadphaseCollision/btSimpleBroadphase.cpp.obj
[ 17%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/SphereTriangleDetector.cpp.obj
[ 18%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btActivatingCollisionAlgorithm.cpp.obj
[ 18%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btBox2dBox2dCollisionAlgorithm.cpp.obj
[ 19%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btBoxBoxCollisionAlgorithm.cpp.obj
[ 19%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btBoxBoxDetector.cpp.obj
[ 19%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp.obj
[ 20%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btCollisionObject.cpp.obj
[ 20%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btCollisionWorld.cpp.obj
[ 20%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btCollisionWorldImporter.cpp.obj
[ 21%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btCompoundCollisionAlgorithm.cpp.obj
[ 21%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btCompoundCompoundCollisionAlgorithm.cpp.obj
[ 22%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btConvex2dConvex2dAlgorithm.cpp.obj
[ 22%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btConvexConcaveCollisionAlgorithm.cpp.obj
[ 22%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btConvexConvexAlgorithm.cpp.obj
[ 23%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btConvexPlaneCollisionAlgorithm.cpp.obj
[ 23%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btDefaultCollisionConfiguration.cpp.obj
[ 23%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btEmptyCollisionAlgorithm.cpp.obj
[ 24%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btGhostObject.cpp.obj
[ 24%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btHashedSimplePairCache.cpp.obj
[ 24%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btInternalEdgeUtility.cpp.obj
[ 25%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btManifoldResult.cpp.obj
[ 25%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btSimulationIslandManager.cpp.obj
[ 26%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btSphereBoxCollisionAlgorithm.cpp.obj
[ 26%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btSphereSphereCollisionAlgorithm.cpp.obj
[ 26%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btSphereTriangleCollisionAlgorithm.cpp.obj
[ 27%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionDispatch/btUnionFind.cpp.obj
[ 27%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btBox2dShape.cpp.obj
[ 27%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btBoxShape.cpp.obj
[ 28%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btBvhTriangleMeshShape.cpp.obj
[ 28%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btCapsuleShape.cpp.obj
[ 29%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btCollisionShape.cpp.obj
[ 29%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btCompoundShape.cpp.obj
[ 29%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConcaveShape.cpp.obj
[ 30%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConeShape.cpp.obj
[ 30%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvex2dShape.cpp.obj
[ 30%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvexHullShape.cpp.obj
[ 31%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvexInternalShape.cpp.obj
[ 31%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvexPointCloudShape.cpp.obj
[ 32%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvexPolyhedron.cpp.obj
[ 32%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvexShape.cpp.obj
[ 32%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btConvexTriangleMeshShape.cpp.obj
[ 33%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btCylinderShape.cpp.obj
[ 33%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btEmptyShape.cpp.obj
[ 33%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btHeightfieldTerrainShape.cpp.obj
[ 34%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btMinkowskiSumShape.cpp.obj
[ 34%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btMultiSphereShape.cpp.obj
[ 35%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btMultimaterialTriangleMeshShape.cpp.obj
[ 35%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btOptimizedBvh.cpp.obj
[ 35%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btPolyhedralConvexShape.cpp.obj
[ 36%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btScaledBvhTriangleMeshShape.cpp.obj
[ 36%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btShapeHull.cpp.obj
[ 36%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btSphereShape.cpp.obj
[ 37%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btStaticPlaneShape.cpp.obj
[ 37%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btStridingMeshInterface.cpp.obj
[ 37%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTetrahedronShape.cpp.obj
[ 38%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTriangleBuffer.cpp.obj
[ 38%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTriangleCallback.cpp.obj
[ 39%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTriangleIndexVertexArray.cpp.obj
[ 39%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTriangleIndexVertexMaterialArray.cpp.obj
[ 39%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTriangleMesh.cpp.obj
[ 40%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btTriangleMeshShape.cpp.obj
[ 40%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/CollisionShapes/btUniformScalingShape.cpp.obj
[ 40%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btContactProcessing.cpp.obj
[ 41%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btGImpactBvh.cpp.obj
[ 41%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btGImpactCollisionAlgorithm.cpp.obj
[ 42%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btGImpactQuantizedBvh.cpp.obj
[ 42%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btGImpactShape.cpp.obj
[ 42%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btGenericPoolAllocator.cpp.obj
[ 43%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/btTriangleShapeEx.cpp.obj
[ 43%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/gim_box_set.cpp.obj
[ 43%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/gim_contact.cpp.obj
[ 44%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/gim_memory.cpp.obj
[ 44%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/Gimpact/gim_tri_collision.cpp.obj
[ 45%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btContinuousConvexCollision.cpp.obj
[ 45%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btConvexCast.cpp.obj
[ 45%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btGjkConvexCast.cpp.obj
[ 46%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btGjkEpa2.cpp.obj
[ 46%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btGjkEpaPenetrationDepthSolver.cpp.obj
[ 46%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btGjkPairDetector.cpp.obj
[ 47%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btMinkowskiPenetrationDepthSolver.cpp.obj
[ 47%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btPersistentManifold.cpp.obj
[ 48%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btPolyhedralContactClipping.cpp.obj
[ 48%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btRaycastCallback.cpp.obj
[ 48%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btSubSimplexConvexCast.cpp.obj
[ 49%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletCollision/NarrowPhaseCollision/btVoronoiSimplexSolver.cpp.obj
[ 49%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Character/btKinematicCharacterController.cpp.obj
[ 49%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btConeTwistConstraint.cpp.obj
[ 50%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btContactConstraint.cpp.obj
[ 50%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btFixedConstraint.cpp.obj
[ 51%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btGearConstraint.cpp.obj
[ 51%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btGeneric6DofConstraint.cpp.obj
[ 51%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btGeneric6DofSpring2Constraint.cpp.obj
[ 52%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btGeneric6DofSpringConstraint.cpp.obj
[ 52%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btHinge2Constraint.cpp.obj
[ 52%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btHingeConstraint.cpp.obj
[ 53%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btNNCGConstraintSolver.cpp.obj
[ 53%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btPoint2PointConstraint.cpp.obj
[ 53%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btSequentialImpulseConstraintSolver.cpp.obj
[ 54%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btSliderConstraint.cpp.obj
[ 54%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btSolve2LinearConstraint.cpp.obj
[ 55%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btTypedConstraint.cpp.obj
[ 55%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/ConstraintSolver/btUniversalConstraint.cpp.obj
[ 55%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp.obj
[ 56%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Dynamics/btRigidBody.cpp.obj
[ 56%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Dynamics/btSimpleDynamicsWorld.cpp.obj
[ 56%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBody.cpp.obj
[ 57%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBodyConstraint.cpp.obj
[ 57%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBodyConstraintSolver.cpp.obj
[ 58%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp.obj
[ 58%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBodyJointLimitConstraint.cpp.obj
[ 58%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBodyJointMotor.cpp.obj
[ 59%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Featherstone/btMultiBodyPoint2Point.cpp.obj
[ 59%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/MLCPSolvers/btDantzigLCP.cpp.obj
[ 59%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/MLCPSolvers/btLemkeAlgorithm.cpp.obj
[ 60%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/MLCPSolvers/btMLCPSolver.cpp.obj
[ 60%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Vehicle/btRaycastVehicle.cpp.obj
[ 61%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletDynamics/Vehicle/btWheelInfo.cpp.obj
[ 61%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btDefaultSoftBodySolver.cpp.obj
[ 61%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftBody.cpp.obj
[ 62%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftBodyConcaveCollisionAlgorithm.cpp.obj
[ 62%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftBodyHelpers.cpp.obj
[ 62%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftBodyRigidBodyCollisionConfiguration.cpp.obj
[ 63%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftRigidCollisionAlgorithm.cpp.obj
[ 63%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftRigidDynamicsWorld.cpp.obj
[ 64%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/BulletSoftBody/btSoftSoftCollisionAlgorithm.cpp.obj
[ 64%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btAlignedAllocator.cpp.obj
[ 64%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btConvexHull.cpp.obj
[ 65%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btConvexHullComputer.cpp.obj
[ 65%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btGeometryUtil.cpp.obj
[ 65%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btPolarDecomposition.cpp.obj
[ 66%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btQuickprof.cpp.obj
[ 66%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btSerializer.cpp.obj
[ 66%] Building CXX object extern/bullet/CMakeFiles/bullet.dir/LinearMath/btVector3.cpp.obj
[ 67%] Linking CXX static library libbullet.a
[ 67%] Built target bullet
Scanning dependencies of target OpenTomb
[ 67%] Building C object CMakeFiles/OpenTomb.dir/src/core/avl.c.obj
[ 67%] Building C object CMakeFiles/OpenTomb.dir/src/core/base_types.c.obj
[ 68%] Building C object CMakeFiles/OpenTomb.dir/src/core/console.c.obj
[ 68%] Building C object CMakeFiles/OpenTomb.dir/src/core/gl_font.c.obj
[ 68%] Building C object CMakeFiles/OpenTomb.dir/src/core/gl_text.c.obj
[ 69%] Building C object CMakeFiles/OpenTomb.dir/src/core/gl_util.c.obj
[ 69%] Building C object CMakeFiles/OpenTomb.dir/src/core/obb.c.obj
[ 69%] Building C object CMakeFiles/OpenTomb.dir/src/core/polygon.c.obj
[ 70%] Building C object CMakeFiles/OpenTomb.dir/src/core/system.c.obj
[ 70%] Building C object CMakeFiles/OpenTomb.dir/src/core/utf8_32.c.obj
[ 71%] Building C object CMakeFiles/OpenTomb.dir/src/core/vmath.c.obj
[ 71%] Building CXX object CMakeFiles/OpenTomb.dir/src/gui/gui.cpp.obj
[ 71%] Building CXX object CMakeFiles/OpenTomb.dir/src/gui/gui_inventory.cpp.obj
[ 72%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/tiny_codec.c.obj
[ 72%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/stream_codec.c.obj
[ 72%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/containers/rpl.c.obj
[ 73%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/codecs/adpcm.c.obj
[ 73%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/codecs/adpcm_data.c.obj
[ 74%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/codecs/pcm.c.obj
[ 74%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/codecs/escape124.c.obj
[ 74%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/codecs/escape130.c.obj
[ 75%] Building C object CMakeFiles/OpenTomb.dir/src/fmv/codecs/reverse.c.obj
[ 75%] Building CXX object CMakeFiles/OpenTomb.dir/src/physics/physics_bullet.cpp.obj
[ 75%] Building CXX object CMakeFiles/OpenTomb.dir/src/physics/hair.cpp.obj
[ 76%] Building CXX object CMakeFiles/OpenTomb.dir/src/physics/ragdoll.cpp.obj
[ 76%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/bordered_texture_atlas.cpp.obj
[ 77%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/bsp_tree.cpp.obj
[ 77%] Building C object CMakeFiles/OpenTomb.dir/src/render/bsp_tree_2d.c.obj
[ 77%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/camera.cpp.obj
[ 78%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/frustum.cpp.obj
[ 78%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/render.cpp.obj
[ 78%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/shader_description.cpp.obj
[ 79%] Building CXX object CMakeFiles/OpenTomb.dir/src/render/shader_manager.cpp.obj
[ 79%] Building CXX object CMakeFiles/OpenTomb.dir/src/script/script.cpp.obj
[ 80%] Building CXX object CMakeFiles/OpenTomb.dir/src/script/script_audio.cpp.obj
[ 80%] Building CXX object CMakeFiles/OpenTomb.dir/src/script/script_character.cpp.obj
[ 80%] Building CXX object CMakeFiles/OpenTomb.dir/src/script/script_entity.cpp.obj
[ 81%] Building CXX object CMakeFiles/OpenTomb.dir/src/script/script_skeletal_model.cpp.obj
[ 81%] Building CXX object CMakeFiles/OpenTomb.dir/src/script/script_world.cpp.obj
[ 81%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_common.cpp.obj
[ 82%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_main.cpp.obj
[ 82%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_tr1.cpp.obj
[ 82%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_tr2.cpp.obj
[ 83%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_tr3.cpp.obj
[ 83%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_tr4.cpp.obj
[ 84%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/l_tr5.cpp.obj
[ 84%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/scaler.cpp.obj
[ 84%] Building CXX object CMakeFiles/OpenTomb.dir/src/vt/vt_level.cpp.obj
[ 85%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control.cpp.obj
[ 85%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_Lara.cpp.obj
[ 85%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_bat.cpp.obj
[ 86%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_wolf.cpp.obj
[ 86%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_bear.cpp.obj
[ 87%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_raptor.cpp.obj
[ 87%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_trex.cpp.obj
[ 87%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_Larson.cpp.obj
[ 88%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_Pierre.cpp.obj
[ 88%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_lion.cpp.obj
[ 88%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_gorilla.cpp.obj
[ 89%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_crocodile.cpp.obj
[ 89%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_rat.cpp.obj
[ 90%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_centaur.cpp.obj
[ 90%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_puma.cpp.obj
[ 90%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_winged_mutant.cpp.obj
[ 91%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_MrT.cpp.obj
[ 91%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_cowboy.cpp.obj
[ 91%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_skateboardist.cpp.obj
[ 92%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_torso_boss.cpp.obj
[ 92%] Building CXX object CMakeFiles/OpenTomb.dir/src/state_control/state_control_Natla.cpp.obj
[ 93%] Building CXX object CMakeFiles/OpenTomb.dir/src/audio/audio.cpp.obj
[ 93%] Building CXX object CMakeFiles/OpenTomb.dir/src/audio/audio_fx.cpp.obj
[ 93%] Building CXX object CMakeFiles/OpenTomb.dir/src/audio/audio_stream.cpp.obj
[ 94%] Building C object CMakeFiles/OpenTomb.dir/src/audio/stb_vorbis.c.obj
/home/Carlo/opentomb/src/audio/stb_vorbis.c:580:0: warning: "alloca" redefined
    #define alloca __builtin_alloca

In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/stdlib.h:695:0,
                 from /home/Carlo/opentomb/src/audio/stb_vorbis.c:549:
/usr/i686-w64-mingw32/sys-root/mingw/include/malloc.h:183:0: note: this is the location of the previous definition
 #define alloca(x) __builtin_alloca((x))

[ 94%] Building CXX object CMakeFiles/OpenTomb.dir/src/character_controller.cpp.obj
[ 94%] Building CXX object CMakeFiles/OpenTomb.dir/src/controls.cpp.obj
[ 95%] Building CXX object CMakeFiles/OpenTomb.dir/src/engine.cpp.obj
[ 95%] Building CXX object CMakeFiles/OpenTomb.dir/src/entity.cpp.obj
[ 95%] Building CXX object CMakeFiles/OpenTomb.dir/src/game.cpp.obj
[ 96%] Building CXX object CMakeFiles/OpenTomb.dir/src/game_camera.cpp.obj
[ 96%] Building CXX object CMakeFiles/OpenTomb.dir/src/gameflow.cpp.obj
[ 97%] Building CXX object CMakeFiles/OpenTomb.dir/src/inventory.cpp.obj
[ 97%] Building CXX object CMakeFiles/OpenTomb.dir/src/image.cpp.obj
[ 97%] Building CXX object CMakeFiles/OpenTomb.dir/src/main_SDL.cpp.obj
[ 98%] Building C object CMakeFiles/OpenTomb.dir/src/mesh.c.obj
[ 98%] Building CXX object CMakeFiles/OpenTomb.dir/src/resource.cpp.obj
[ 98%] Building CXX object CMakeFiles/OpenTomb.dir/src/room.cpp.obj
[ 99%] Building C object CMakeFiles/OpenTomb.dir/src/skeletal_model.c.obj
[ 99%] Building CXX object CMakeFiles/OpenTomb.dir/src/trigger.cpp.obj
[100%] Building CXX object CMakeFiles/OpenTomb.dir/src/world.cpp.obj
[100%] Building RC object CMakeFiles/OpenTomb.dir/resource/icon/opentomb.rc.res
[100%] Linking CXX executable OpenTomb.exe
[100%] Built target OpenTomb

Carlo@CarloPC ~/ot
$
```

I always got that warning into src/audio/stb_vorbis.c, perhaps it had been originally done with a not so recent version of W32API package. In the doubt, the best solution would add an ```#undef alloca``` before defining it again. In that way, it should not appear anymore.